frontend/src/scalars/ui/Chart.vue

```diff
@@ -0,0 +1,598 @@
+            this.initDownloadType();
+        },
+        originData: function(val) {
+            this.setChartData();
+            this.setChartsOutlier();
+            this.setChartHorizon();
+        },
+        smoothing: function(val) {
+            this.setChartData();
+        },
+        outlier: function(val) {
+            this.setChartsOutlier();
+        },
+        horizontal: function(val) {
+            this.setChartHorizon();
+        },
+        // runs: function(val) {
+        //     this.setChartsRuns();
+        // }
+    },
+    mounted() {
+        this.initChart(this.tagInfo);
+
+        if (this.running) {
+            this.startInterval();
+        }
+
+        this.$watch('running', function(running) {
+            running ? this.startInterval() : this.stopInterval();
+        });
+    },
+
+    beforeDestroy() {
+        this.stopInterval();
+    },
+    methods: {
+        initDownloadType() {
+            if (this.runsItems.length === 0) {
+                return;
+            }
+            this.downloadType = this.runsItems[0].value;
+        },
+
+        // Create a Scalar Chart, initialize it with default settings, then load datas
+        initChart(tagInfo) {
+            this.createChart();
+            this.setChartsOptions(tagInfo);
+            this.getOriginChartData(tagInfo);
+        },
+
+        createChart() {
+            let el = document.getElementsByClassName('visual-dl-chart-box')[0];
+            this.myChart = echarts.init(el);
+        },
+
+        setChartsOptions({tagList, tag}) {
+            // Create two lines, one line is original, the other line is for smoothing
+            let seriesOption = tagList.map(item => [
+                    {
+                        name: item.run,
+                        type: 'line',
+                        showSymbol: false,
+                        hoverAnimation: false,
+                        z: 0,
+                        data: [],
+                        animationDuration: 100,
+                        lineStyle: {
+                            normal: {
+                                opacity: originLinesOpacity,
+                                width: lineWidth
+                            }
+                        }
+                    },
+                    {
+                        name: item.run,
+                        type: 'line',
+                        showSymbol: false,
+                        hoverAnimation: false,
+                        z: 1,
+                        data: [],
+                        animationDuration: 100,
+                        lineStyle: {
+                            normal: {
+                                width: lineWidth
+                            }
+                        }
+                    }
+                ]
+            );
+            seriesOption = flatten(seriesOption);
+            let legendOptions = tagList.map(item => item.run);
+            let that = this;
```

Comment:
let view = this; Who came up with `that`

Reply:
lol, I coped all these from original san files